### PR TITLE
SONARIAC-878 Refactor assertions to split ExpressionAssert

### PR DESCRIPTION
**Pull request by rudy-regazzoni-sonarsource**

### CODE REVIEW

File: iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/api/HasTextRangeAssert.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.arm.tree.api;
+
+import org.sonar.iac.common.testing.TextRangeAssert;
+
+public class HasTextRangeAssert<SELF extends ArmTreeAssert<SELF, ACTUAL>, ACTUAL extends ArmTree> extends ArmTreeAssert<SELF, ACTUAL> {
```
Comment (mstachniuk) [inline iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/api/HasTextRangeAssert.java:24]: IMO this class can be abstract

File: iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/api/ObjectExpressionAssert.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import org.sonar.iac.common.checks.PropertyUtils;
+
+import static org.sonar.iac.arm.tree.api.ExpressionAssert.assertThat;
+
+public class ObjectExpressionAssert extends HasTextRangeAssert<ObjectExpressionAssert, ObjectExpression> {
+  protected ObjectExpressionAssert(ObjectExpression objectExpression) {
+    super(objectExpression, ObjectExpressionAssert.class);
+  }
+
+  public ObjectExpressionAssert hasValue(String key, String value) {
```
Comment (mstachniuk) [inline iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/api/ObjectExpressionAssert.java:32]: Maybe `containsProperty` or `containsKeyValue` will be a better name?

File: iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/api/ArrayExpressionAssert.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.arm.tree.api;
+
+import org.assertj.core.api.Assertions;
+
+import static org.sonar.iac.arm.tree.api.ExpressionAssert.assertThat;
+
+public class ArrayExpressionAssert extends HasTextRangeAssert<ArrayExpressionAssert, ArrayExpression> {
+  protected ArrayExpressionAssert(ArrayExpression arrayExpression) {
```
Comment (mstachniuk) [inline iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/api/ArrayExpressionAssert.java:27]: The constructor can be private. The same in other classes (but without `HasTextRangeAssert`)

File: iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/api/BooleanLiteralAssert.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+package org.sonar.iac.arm.tree.api;
+
+import org.assertj.core.api.Assertions;
+
+public class BooleanLiteralAssert extends HasTextRangeAssert<BooleanLiteralAssert, BooleanLiteral> {
+  protected BooleanLiteralAssert(BooleanLiteral booleanLiteral) {
+    super(booleanLiteral, BooleanLiteralAssert.class);
+  }
+
+  public BooleanLiteralAssert hasValue(boolean value) {
```
Comment (mstachniuk) [inline iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/api/BooleanLiteralAssert.java:29]: Consider `isTrue()` & `isFalse()` methods instead to follow AssertJ conventions

**Review by mstachniuk — CHANGES_REQUESTED** [review]
Good Job and please see my comments, some of them are important

File: iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/api/ArrayExpressionAssert.java (hunk 3 above)
Comment (mstachniuk) [reply iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/api/ArrayExpressionAssert.java]: There is `assertThat()` method for that, so the constructors can be private

**Review by mstachniuk — APPROVED** [review]
LGTM and please consider using `assertThat()` method instead of package private constructors.

File: iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/api/ArrayExpressionAssert.java (hunk 3 above)
Comment (rudy-regazzoni-sonarsource) [reply iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/api/ArrayExpressionAssert.java]: Make sense, will do that 👍

**Pull request merged**